tests/S3/S3ClientTest.php

```diff
@@ -79,6 +79,19 @@ public function testValidatesDnsBucketNames($bucket, $valid)
         $this->assertEquals($valid, S3Client::isBucketDnsCompatible($bucket));
     }
 
+    public function testValidateDnsBucketNamesPassingNull()
```

Comment:
This test passes with or without the above code update

Reply:
Which PHP version did you test with?

Reply:
Though personally I'd say this doesn't necessarily need a test; the code is pretty self explanatory- anyone with passing knowledge of PHP 8 will understand why we put the `?:` there and know not to remove it.  The functionality hasn't changed, so a unit test isn't necessarily doing anything here. That said, this change is included in this PR https://github.com/aws/aws-sdk-php/pull/2403/files, so do we want to merge it?